nekoyume/Assets/_Scripts/State/States.cs

```diff
@@ -511,7 +508,12 @@ private static async UniTask<AvatarState> GetAvatarStateAsync(Address address, b
                 _avatarStates.Add(index, state);
             }
 
-            await InitItemSlotState(index, state);
+            if (index == CurrentAvatarKey)
+            {
+                await InitRuneStoneBalanceAsync();
+                await InitRuneStatesAsync();
+            }
```

Comment:
Key difference is here. RuneState and RuneStoneBalance is updated on every `AvatarState` updates.